### PR TITLE
Debug build Bug fixes

### PR DESCRIPTION
**Pull request by pcoramasionwu**
Resolve issues that prevented building tf in debug mode

### CODE REVIEW

File: tensorflow/core/framework/op_def_builder.cc [hunk 1]
```diff
@@ -114,14 +114,16 @@ bool ConsumeAttrNumber(StringPiece* sp, int64_t* out) {
...
+  const char* capture_data = sp->data();
+  size_t initial_length = sp->length(); 
+
   if (absl::ConsumePrefix(sp, "numbertype") ||
       absl::ConsumePrefix(sp, "numerictype") ||
       absl::ConsumePrefix(sp, "quantizedtype") ||
       absl::ConsumePrefix(sp, "realnumbertype") ||
       absl::ConsumePrefix(sp, "realnumberictype")) {
-    *out = StringPiece(capture_data, sp->begin() - capture_begin);
+    size_t consumed_length = initial_length - sp->length(); // prevent error: 'expression cannot subtract incompatible string_view iterators'
```
Comment (pcoramasionwu) [inline tensorflow/core/framework/op_def_builder.cc:125]: subtract lengths instead of iterators

File: tensorflow/core/kernels/depthtospace_op.cc [hunk 2]
```diff
@@ -173,6 +173,35 @@ struct DepthToSpaceOpFunctor<CPUDevice, T, FORMAT_NHWC> {
     }
   }
 };
+
+template <typename T>
```
Comment (pcoramasionwu) [inline tensorflow/core/kernels/depthtospace_op.cc:177]: add missing template funtion- not sure how this compiled in release mode...

File: tensorflow/core/kernels/spacetodepth_op.cc [hunk 3]
```diff
@@ -189,6 +189,34 @@ struct SpaceToDepthOpFunctor<CPUDevice, T, FORMAT_NHWC> {
     }
   }
 };
+
+template <typename T>
```
Comment (pcoramasionwu) [inline tensorflow/core/kernels/spacetodepth_op.cc:193]: add missing template function

File: tensorflow/core/framework/op_def_builder.cc (hunk 1 above)
Comment (alainslak) [reply tensorflow/core/framework/op_def_builder.cc:125]: Are these params in the correct order? They appear to be swapped from the original implementation where it looks like it is doing current - previous and you are doing previous - current.

Comment (pcoramasionwu) [reply tensorflow/core/framework/op_def_builder.cc:125]: it's a bit confusing but we are calculation the length of the consumed portion of the string using two different approaching.

the original implementation is doing **current position - original position**
I'm doing **initial length - current length**

both work out to the length of the consumed portion

Comment (alainslak) [reply tensorflow/core/framework/op_def_builder.cc:125]: I don't really get how that works without taking the absolute value or something, but if you're confident that it's giving you what you want, then it's all good.

**Review by alainslak — APPROVED** [review]
LGTM

File: tensorflow/core/framework/op_def_builder.cc (hunk 1 above)
Comment (poramasionwu) [reply tensorflow/core/framework/op_def_builder.cc:125]: I'm going to test the logic to make sure it checks out... it's worth double-checking.

Comment (poramasionwu) [reply tensorflow/core/framework/op_def_builder.cc:125]: seems to check out
image

**Pull request merged**